File: C2109L/ADF1/20-09-2022/javaPE01/src/main/java/com/aptech/inheritance/Circle.java
package com.aptech.inheritance;

public class Circle extends Shape{
    private int radius;

    public Circle(String name, String color, int radius) {
        super(name, color);
        this.radius = radius;
    }

    @Override
    public int getArea() {
        return (int)Math.PI * radius * radius;//explicit casting
    }
    public double getFloatArea() {
        return Math.PI * radius * radius;//explicit casting
    }

    @Override
    public String toString() {
        return super.toString() + "\n"+
                "radius=" + radius + "\n"+
                "area = "+this.getArea();

    }
}

File: C2110I/Dart/myflutterapp/lib/models/circle.dart
import 'dart:math';

import 'package:flutter/cupertino.dart';

class Circle {
  final double radius;
  Circle({
    this.radius = 1.0,
  });
  get perimeter => 3.14 * radius * 2;
  get area => pi * radius * radius;
}